cmake/kokkos_arch.cmake

```diff
@@ -613,6 +613,17 @@ IF (KOKKOS_ENABLE_HIP)
       DEFAULT -fno-gpu-rdc
     )
   ENDIF()
+
+  set(KOKKOS_IMPL_HAVE_HIP_WORKING_GRAPH_CODE "
+    #include <hip/hip_runtime_api.h>
+
+    #if HIP_VERSION_MAJOR == 5 && HIP_VERSION_MINOR == 2
+      #error \"HIP does not have a working implementation of 'hipGraph'.\"
+    #endif
+    int main() {}
+  ")
+  check_cxx_source_runs(KOKKOS_IMPL_HAVE_HIP_WORKING_GRAPH_CODE KOKKOS_IMPL_HAVE_HIP_WORKING_GRAPH_IMPLEMENTATION)
+  unset(KOKKOS_IMPL_HAVE_HIP_WORKING_GRAPH_CODE)
```

Comment:
Why are you not using `KOKKOS_CXX_COMPILER_VERSION`?

Reply:
The only doubt I have: is it true that `hipcc` is the only compiler supported by `Kokkos` to compile `HIP` ?

Reply:
@Rombur

Reply:
>  is it true that hipcc is the only compiler supported by Kokkos to compile HIP ?

No, we also support `amdclang` and Cray Clang

Reply:
OK, so the proposed `if` condition based on the `HIPCC` compiler version would not be generic, right ?

I guess we've got 2 choices:
- be pragmatic and say that the `if` will discard the test for `hipcc < 5.3` **and** if the compiler is not `hipcc`
- come up with another way of detecting the version of `HIP`

Reply:
You could do something like this in the test itself
```C++
#if ((HIP_VERSION_MAJOR == 5) && (HIP_VERSION_MINOR == 2))
GTEST_SKIP();
#else 
Your test
#endif
```

Reply:
Yeah I know, but I really don't want to add a unit test executable whose all tests are skipped. It makes people think it is tested, until they carefully look at the test output (which no one does when `ctest` reports that all is fine).

Reply:
I think we can close this one too.